### PR TITLE
fixes #3440 "Campaign notification is not removed from Home even when disabling it in the setting"

### LINKED ISSUE #3440: Campaign notification is not removed from Home even after disabling it in the Settings
**Summary:** 

Campaign notification is not removed from Home even after disabling the "Display campaigns" preference in the Settings. This happens because the `MainActivity` is in memory and is not being recreated.

**How can we reproduce the issue?** 

Open Home. 
Notice the campaign notification. 
Go to Settings and disable "Display campaigns". 
Go back to Home.

**What did you expect the app to do, and what did you see instead?**

I expected the campaigns notification to have been removed when I reopened Home.

**Device and Android version:** 

Redmi Note 5 Pro
MIUI version 11
Android 9
 
**Commons app version:** 

2.12.0-debug-master~d0d663578
master and betaDebug

**Screen-shots:** 

del

**Would you like to work on the issue?**

Yes, if we can agree the issue is genuine then I'd like to work on it.

### PR DESCRIPTION
**Pull request by 4D17Y4**
**Description (required)**

Fixes #3440

What changes did you make and why?

Added missing else statement while fetching campaign.
Swipe to disable notification enabled.

**Tests performed (required)**

Tested betadebug on Nokia 6.1+ android 10 stock.

### CODE REVIEW

**Comment by 4D17Y4** [comment]
up for review @nicolas-raoul, @neslihanturan.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java [hunk 1]
```diff
@@ -572,16 +572,17 @@ private void fetchCampaigns() {
...
+            campaignView.setVisibility(View.GONE);
+        }
     }
 
     @Override public void showMessage(String message) {
         Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
     }
 
     @Override public void showCampaigns(Campaign campaign) {
-        if (campaign != null) {
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:582]: The PR seems right in general, thanks @4D17Y4 :) But why did you removed this null check? I think it should stay.

**Review by neslihanturan — CHANGES_REQUESTED** [review]
Just a minor change :)

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java (hunk 1 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:582]: Actually the null chek is handled on function call below. It sets the visibility of the campaign view to gone if the campaign is null. So I don't think we need it.

Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:582]: https://github.com/commons-app/apps-android-commons/blob/ac297e42f601e88f1fe24969822cb76762a13e8b/app/src/main/java/fr/free/nrw/commons/campaigns/CampaignView.java#L48-L56

This is the mentioned method.
Also, if the campaign is null the condition check will bypass the functions check and the campaignView visibility would still be visible which may cause some unexpected errors.

I was on another device so could not reference the code in the above comment. ;)

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java:582]: For now, you are right it seems redundant. However, I think having the null check inside the method will be a better practice. Some people may use showCampaign method without nullcheck in future. So I think we should keep it inside the method :)

**Review by neslihanturan — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java (hunk 1 above)
Comment (4D17Y4) [reply app/src/main/java/fr/free/nrw/commons/contributions/ContributionsFragment.java]: I get it @neslihanturan,
Sorry for trouble,
I've made the required changes
Thanks.

**Review by neslihanturan — APPROVED** [review]
Thanks @4D17Y4 :)

**Pull request merged**